src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -885,7 +885,7 @@ private static MethodBodyStatement CreatePropertyNullCheckStatement(JsonProperty
         private static BoolExpression GetCheckEmptyPropertyValueExpression(JsonPropertyExpression jsonProperty, JsonPropertySerialization jsonPropertySerialization, bool shouldTreatEmptyStringAsNull)
         {
             var jsonElement = jsonProperty.Value;
-            if (!shouldTreatEmptyStringAsNull)
+            if (!shouldTreatEmptyStringAsNull && !jsonPropertySerialization.DeserializeEmptyStringAsNull)
```

Comment:
So, `jsonPropertySerialization.DeserializeEmptyStringAsNull` is based on the property decorator, which only deal with TypeSpec inputs.
And `shouldTreatEmptyStringAsNull` is the result from configuration of swagger inputs.
I tend to separate the usage here. If `jsonPropertySerialization.DeserializeEmptyStringAsNull` is true, we should just return L903 regardless the other conditions.
We can add a test case which doesn't meet L898.

In L898,